libs/feature/search/src/lib/utils/links/link-utils.ts

```
@@ -47,17 +50,16 @@ export function getLinksWithWfsFormats(
   link: MetadataLinkValid
 ): Promise<MetadataLinkValid[]> {
   return new WfsEndpoint(link.url).isReady().then((endpoint) => {
-    let featureType = endpoint.getFeatureTypeSummary(link.name)
-    if (!featureType) {
-      //fallback: if link name does not match layer name, use first layer
-      featureType = endpoint.getFeatureTypes()[0]
-      featureType.outputFormats = endpoint.getServiceInfo().outputFormats
+    const featureType = endpoint.getFeatureTypeSummary(link.name)
+    if (featureType) {
+      return featureType.outputFormats.map((format) => ({
+        ...link,
+        url: endpoint.getFeatureUrl(featureType.name, { outputFormat: format }),
+        format: format,
+      }))
+    } else {
+      return Promise.reject(new Error('downloads.wfs.featuretype.not.found'))
```

you can probably just throw the error here, less confusing I think